src/installer/tests/HostActivation.Tests/SymbolicLinks.cs

```diff
@@ -21,6 +23,90 @@ public SymbolicLinks(SymbolicLinks.SharedTestState fixture)
             sharedTestState = fixture;
         }
 
+        [Theory]
+        [InlineData("a/b/SymlinkToFrameworkDependentApp")]
+        [InlineData("a/SymlinkToFrameworkDependentApp")]
+        public void Symlink_all_files_fx(string symlinkRelativePath)
+        {
+            using var testDir = TestArtifact.Create("symlink");
+            Directory.CreateDirectory(Path.Combine(testDir.Location, Path.GetDirectoryName(symlinkRelativePath)));
+
+            // Symlink every file in the app directory
+            var symlinks = new List<SymLink>();
+            try
+            {
+                foreach (var file in Directory.EnumerateFiles(sharedTestState.FrameworkDependentApp.Location))
+                {
+                    var fileName = Path.GetFileName(file);
+                    var symlinkPath = Path.Combine(testDir.Location, symlinkRelativePath, fileName);
+                    Directory.CreateDirectory(Path.GetDirectoryName(symlinkPath));
+                    symlinks.Add(new SymLink(symlinkPath, file));
+                }
+
+                var result = Command.Create(Path.Combine(testDir.Location, symlinkRelativePath, Path.GetFileName(sharedTestState.FrameworkDependentApp.AppExe)))
+                    .CaptureStdErr()
+                    .CaptureStdOut()
+                    .DotNetRoot(TestContext.BuiltDotNet.BinPath)
+                    .Execute();
+
+                // This should succeed on all platforms, but for different reasons:
+                // * Windows: The apphost will look next to the symlink for the app dll and find the symlinked dll
+                // * Unix: The apphost will look next to the resolved apphost for the app dll and find the real thing
```

Comment:
Maybe split this into separate cases? Basically so that the Windows case would fail without this revert.
- Windows: app files do not exist at the target directory, only in the symlink directory
- Unix: app files exist at the target directory

Reply:
I'm not sure I understand the Windows case. Are you suggesting that in the Windows case we should separate out the files behind the symlink? It seems like that would be the only way it would fail without the revert.

Reply:
Yeah, separate out just the apphost. I'm looking to hit the case that broke for folks - with the current test, Windows would still pass, since the files can still be resolved after following the apphost symlink since they are at the target.

For Windows, I'm thinking something like:
```
symlinked\
  app.exe -> target_app\app.exe
  app.dll -> app_deps\app.dll
  (other dependencies) -> app_deps\(other dependencies)

target_app\
  app.exe
  (no other files)

app_deps\
  app.dll
  (other dependencies)
```